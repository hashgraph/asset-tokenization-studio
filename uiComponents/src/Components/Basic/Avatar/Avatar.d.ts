import { type avatarAnatomy as ChakraAvatarParts } from "@chakra-ui/anatomy";
import { type AvatarProps as ChakraAvatarProps } from "@chakra-ui/avatar";
import type { SystemStyleObject as ChakraSystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import type { PartsStyleInterpolation, BaseMultiStyleConfiguration } from "../../../Theme/Components/BaseMultiStyleConfiguration";
export declare const avatarPartsList: typeof ChakraAvatarParts.keys;
type Parts = typeof avatarPartsList;
export interface AvatarThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (({ name, badgeColor, }: {
        badgeColor?: string;
        name?: string;
    }) => Partial<AvatarThemeStyle>) | PartsStyleInterpolation<Parts>;
}
type AvatarThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;
export interface AvatarProps extends Omit<ChakraAvatarProps, "children"> {
    showBadge?: boolean;
    badgeColor?: string;
    isLoading?: boolean;
}
export declare const Avatar: ComponentWithAs<"span", AvatarProps>;
export {};
