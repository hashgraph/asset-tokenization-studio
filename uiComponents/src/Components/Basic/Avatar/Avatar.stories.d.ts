import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithImage: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithBadge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithCustomBadgeColor: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithInitialsAndBadge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithImageAndBadge: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithIsLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, any, import("./Avatar").AvatarProps, import("@chakra-ui/system/dist/system.types").As>>;
