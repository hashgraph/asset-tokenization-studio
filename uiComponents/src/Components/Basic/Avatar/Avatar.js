var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Avatar as ChakraAvatar, AvatarBadge as ChakraAvatarBadge, } from "@chakra-ui/avatar";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { SkeletonCircle as ChakraSkeletonCircle } from "@chakra-ui/react";
import { User } from "@phosphor-icons/react";
import React from "react";
import { PhosphorIcon, Weight } from "../../Foundations/PhosphorIcon";
export var avatarPartsList = [
    "container",
    "badge",
    "label",
    "group",
    "excessLabel",
];
export var Avatar = forwardRef(function (_a, ref) {
    var _b = _a.showBadge, showBadge = _b === void 0 ? false : _b, badgeColor = _a.badgeColor, size = _a.size, name = _a.name, src = _a.src, variant = _a.variant, isLoading = _a.isLoading, props = __rest(_a, ["showBadge", "badgeColor", "size", "name", "src", "variant", "isLoading"]);
    var styles = useChakraMultiStyleConfig("Avatar", {
        name: name && !src,
        size: size,
        badgeColor: badgeColor,
        variant: variant
    });
    if (isLoading) {
        return (React.createElement(ChakraSkeletonCircle, __assign({ sx: styles.container, size: size }, props)));
    }
    return (React.createElement(ChakraAvatar, __assign({ ref: ref, name: name, src: src, sx: styles.container, variant: variant, icon: React.createElement(PhosphorIcon, { weight: Weight.Regular, size: "sm", as: User }) }, props), showBadge && React.createElement(ChakraAvatarBadge, { sx: styles.badge })));
});
