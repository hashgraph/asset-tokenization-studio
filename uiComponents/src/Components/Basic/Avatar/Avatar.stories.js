var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Avatar } from "./Avatar";
var image = "https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60";
var meta = {
    title: "Design System/Basic/Avatar",
    component: Avatar,
    args: {
        isLoading: false
    },
    argTypes: {
        size: {
            control: "select",
            options: ["md"]
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1825-13768&t=3ozoVrWQYvgw1fj3-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return React.createElement(Avatar, __assign({}, args)); };
export var Default = Template.bind({});
export var WithImage = Template.bind({});
WithImage.args = {
    src: image
};
export var WithBadge = Template.bind({});
WithBadge.args = {
    showBadge: true
};
export var WithCustomBadgeColor = Template.bind({});
WithCustomBadgeColor.args = {
    showBadge: true,
    badgeColor: "red.500"
};
export var WithInitialsAndBadge = Template.bind({});
WithInitialsAndBadge.args = {
    name: "John Doe",
    showBadge: true
};
export var WithImageAndBadge = Template.bind({});
WithImageAndBadge.args = {
    src: image,
    name: "John Doe",
    showBadge: true
};
export var WithIsLoading = Template.bind({});
WithIsLoading.args = {
    isLoading: true
};
