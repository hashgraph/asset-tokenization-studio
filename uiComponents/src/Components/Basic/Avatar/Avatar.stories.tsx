import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Avatar } from "./Avatar";

const image =
  "https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60";

const meta = {
  title: "Design System/Basic/Avatar",
  component: Avatar,
  args: {
    isLoading: false,
  },
  argTypes: {
    size: {
      control: "select",
      options: ["md"],
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1825-13768&t=3ozoVrWQYvgw1fj3-4",
    },
    docs: {},
  },
} as Meta<typeof Avatar>;
export default meta;

const Template: StoryFn<typeof Avatar> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});

export const WithImage = Template.bind({});

WithImage.args = {
  src: image,
};

export const WithBadge = Template.bind({});

WithBadge.args = {
  showBadge: true,
};

export const WithCustomBadgeColor = Template.bind({});

WithCustomBadgeColor.args = {
  showBadge: true,
  badgeColor: "red.500",
};

export const WithInitialsAndBadge = Template.bind({});

WithInitialsAndBadge.args = {
  name: "John Doe",
  showBadge: true,
};

export const WithImageAndBadge = Template.bind({});

WithImageAndBadge.args = {
  src: image,
  name: "John Doe",
  showBadge: true,
};

export const WithIsLoading = Template.bind({});
WithIsLoading.args = {
  isLoading: true,
};
