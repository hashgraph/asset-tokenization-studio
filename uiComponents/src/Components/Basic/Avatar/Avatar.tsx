import { type avatarAnatomy as ChakraAvatarParts } from "@chakra-ui/anatomy";
import {
  Avatar as ChakraAvatar,
  AvatarBadge as ChakraAvatarBadge,
  type AvatarProps as ChakraAvatarProps,
} from "@chakra-ui/avatar";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import { SkeletonCircle as ChakraSkeletonCircle } from "@chakra-ui/react";
import { User } from "@phosphor-icons/react";
import React from "react";
import type {
  PartsStyleInterpolation,
  BaseMultiStyleConfiguration,
} from "../../../Theme/Components/BaseMultiStyleConfiguration";
import { PhosphorIcon, Weight } from "../../Foundations/PhosphorIcon";

export const avatarPartsList: typeof ChakraAvatarParts.keys = [
  "container",
  "badge",
  "label",
  "group",
  "excessLabel",
];

type Parts = typeof avatarPartsList;

export interface AvatarThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        name,
        badgeColor,
      }: {
        badgeColor?: string;
        name?: string;
      }) => Partial<AvatarThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

type AvatarThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;
export interface AvatarProps extends Omit<ChakraAvatarProps, "children"> {
  showBadge?: boolean;
  badgeColor?: string;
  isLoading?: boolean;
}

export const Avatar: ComponentWithAs<"span", AvatarProps> = forwardRef<
  AvatarProps,
  "span"
>(
  (
    {
      showBadge = false,
      badgeColor,
      size,
      name,
      src,
      variant,
      isLoading,
      ...props
    }: AvatarProps,
    ref
  ) => {
    const styles = useChakraMultiStyleConfig("Avatar", {
      name: name && !src,
      size,
      badgeColor,
      variant,
    }) as AvatarThemeStyle;

    if (isLoading) {
      return (
        <ChakraSkeletonCircle sx={styles.container} size={size} {...props} />
      );
    }

    return (
      <ChakraAvatar
        ref={ref}
        name={name}
        src={src}
        sx={styles.container}
        variant={variant}
        icon={<PhosphorIcon weight={Weight.Regular} size="sm" as={User} />}
        {...props}
      >
        {showBadge && <ChakraAvatarBadge sx={styles.badge} />}
      </ChakraAvatar>
    );
  }
);
