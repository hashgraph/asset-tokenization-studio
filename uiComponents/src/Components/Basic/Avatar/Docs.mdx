import {
  Story,
  Meta,
  Title,
  Description,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as AvatarStories from "./Avatar.stories";

<Meta of={AvatarStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1825-13768&t=3ozoVrWQYvgw1fj3-4"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Avatar.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Avatar: ConfigAvatar,
  },
};
```

```tsx dark
const ConfigAvatar: AvatarThemeConfiguration = {
  parts: avatarPartsList,
  baseStyle: {},
  sizes: {},
  variants: {},
};
```

### Component parts

You have these parts available: `"container" | "badge" | "label" | "group" | "excessLabel"` (`avatarPartsList`)

## Props

<ArgTypes of={AvatarStories} />

## Stories

<Stories title="" />
