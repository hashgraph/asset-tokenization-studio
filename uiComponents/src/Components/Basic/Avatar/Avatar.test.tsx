import { render } from "@testing-library/react";
import React from "react";
import { Avatar, type AvatarProps } from "./Avatar";

describe(`<Avatar />`, () => {
  const showBadge = true;
  const badgeColor = "red.500";
  const name = "John Doe";
  const src = "https://placekitten.com/200/200";
  const size2xs = { size: "2xs" };
  const sizeXs = { size: "xs" };
  const sizeSm = { size: "sm" };
  const sizeMd = { size: "md" };
  const sizeLg = { size: "lg" };
  const sizeXl = { size: "xl" };
  const size2Xl = { size: "2xl" };
  const size3Xl = { size: "3xl" };
  const factoryComponent = (props: AvatarProps) =>
    render(<Avatar {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent({});
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly with size 2xs", () => {
    const component = factoryComponent(size2xs);
    expect(component.asFragment()).toMatchSnapshot("Using size 2xs");
  });
  test("renders correctly with size xs", () => {
    const component = factoryComponent(sizeXs);
    expect(component.asFragment()).toMatchSnapshot("Using size xs");
  });
  test("renders correctly with size sm", () => {
    const component = factoryComponent(sizeSm);
    expect(component.asFragment()).toMatchSnapshot("Using size sm");
  });
  test("renders correctly with size md", () => {
    const component = factoryComponent(sizeMd);
    expect(component.asFragment()).toMatchSnapshot("Using size md");
  });
  test("renders correctly with size lg", () => {
    const component = factoryComponent(sizeLg);
    expect(component.asFragment()).toMatchSnapshot("Using size lg");
  });
  test("renders correctly with size xl", () => {
    const component = factoryComponent(sizeXl);
    expect(component.asFragment()).toMatchSnapshot("Using size xl");
  });

  test("renders correctly with size 2xl", () => {
    const component = factoryComponent(size2Xl);
    expect(component.asFragment()).toMatchSnapshot("Using size 2xl");
  });

  test("renders correctly with size 3xl", () => {
    const component = factoryComponent(size3Xl);
    expect(component.asFragment()).toMatchSnapshot("Using size 3xl");
  });

  test("renders correctly with initials", () => {
    const component = factoryComponent({ name });
    expect(component.asFragment()).toMatchSnapshot("Using initials");
  });

  test("renders correctly with image", () => {
    const component = factoryComponent({ src });
    expect(component.asFragment()).toMatchSnapshot("Using image");
  });
  test("renders correctly with badge default color", () => {
    const component = factoryComponent({ showBadge });
    expect(component.asFragment()).toMatchSnapshot("Using badge default color");
  });

  test("renders correctly with badge custom color", () => {
    const component = factoryComponent({ showBadge, badgeColor });
    expect(component.asFragment()).toMatchSnapshot("Using badge custom color");
  });

  test("renders correctly with initials and badge default color", () => {
    const component = factoryComponent({ name, showBadge });
    expect(component.asFragment()).toMatchSnapshot(
      "Using initials and badge default color"
    );
  });

  test("renders correctly with initials and badge custom color", () => {
    const component = factoryComponent({ name, showBadge, badgeColor });
    expect(component.asFragment()).toMatchSnapshot(
      "Using initials and badge custom color"
    );
  });

  test("renders correctly with image and badge default color", () => {
    const component = factoryComponent({ src, showBadge });
    expect(component.asFragment()).toMatchSnapshot(
      "Using image and badge default color"
    );
  });

  test("renders correctly with image and badge custom color", () => {
    const component = factoryComponent({ src, showBadge, badgeColor });
    expect(component.asFragment()).toMatchSnapshot(
      "Using image and badge custom color"
    );
  });

  test("renders correctly with isLoading", () => {
    const component = factoryComponent({ isLoading: true });
    expect(component.asFragment()).toMatchSnapshot("Using isLoading");
  });
});
