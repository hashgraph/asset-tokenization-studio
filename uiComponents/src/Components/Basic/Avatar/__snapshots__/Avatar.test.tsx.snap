// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders correctly 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with badge custom color: Using badge custom color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with badge default color: Using badge default color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with image and badge custom color: Using image and badge custom color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with image and badge default color: Using image and badge default color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with image: Using image 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with initials and badge custom color: Using initials and badge custom color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <div
      aria-label="John Doe"
      class="chakra-avatar__initials css-0"
      role="img"
    >
      JD
    </div>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with initials and badge default color: Using initials and badge default color 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <div
      aria-label="John Doe"
      class="chakra-avatar__initials css-0"
      role="img"
    >
      JD
    </div>
    <div
      class="chakra-avatar__badge css-j2wqwy"
    />
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with initials: Using initials 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <div
      aria-label="John Doe"
      class="chakra-avatar__initials css-0"
      role="img"
    >
      JD
    </div>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with isLoading: Using isLoading 1`] = `
<DocumentFragment>
  <div
    class="chakra-skeleton css-1kgvji0"
  />
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size 2xl: Using size 2xl 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size 2xs: Using size 2xs 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size 3xl: Using size 3xl 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size lg: Using size lg 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size md: Using size md 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size sm: Using size sm 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size xl: Using size xl 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`<Avatar /> renders correctly with size xs: Using size xs 1`] = `
<DocumentFragment>
  <span
    class="chakra-avatar css-1ruhx4w"
  >
    <svg
      aria-label=" avatar"
      class="chakra-icon css-13otjrl"
      fill="currentColor"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
